Wod bgc anna

katavouta opened the pull request:
This is try to read WOD. 
Files changed: profile.py where I added the reshape function.

in config/ a new example_WOD_profiles.json to associated with WOD

in examples a new example_scripts a new example to test WOD against SEAsia etc. : example_WOD_BGC_ragged.py

Everything you need to run the example is here (I will move the necessary *.nc file to apporpiate location when finalise everything else): /scratch/accord/COAST/coast_demo/COAsT_example_files/EXTRA_examples/

katavouta commented:
Updated the paths in the example files and added a simple unit test

jpolton reviewed (commented):
You need to add `example_WOD_BGC_ragged.py` to `unit_testing/test_example_scripts.py`

jpolton commented:
Running unit_testing I got lots of good stuff followed by:
```
ok
test_tidetable_tutorial (test_example_scripts.test_example_scripts) ... load and plot HLW data
Tidegauge object at 0x5565df89ff60 initialised
Try the TideGauge.show() method:
Try the TideGauge.get_tidetabletimes() methods:
LT: 2.83 m at 2020-10-13T14:36:00.000000000
HT: 8.01 m at 2020-10-13T07:59:00.000000000
HT: 8.01 m at 2020-10-13T07:59:00.000000000
ok
test_transect_tutorial (test_example_scripts.test_example_scripts) ... ok
test_wcssp_india_example_plot (test_example_scripts.test_example_scripts) ... ERROR
test_load_WOD (test_WOD_read_data.test_WOD_read_data) ... test_reshape_WOD (test_WOD_read_data.test_WOD_read_data) ... ERROR

======================================================================
ERROR: test_amm15_example_plot (test_example_scripts.test_example_scripts)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vkamino/work/jelt/GitHub/COAsT/unit_testing/test_example_scripts.py", line 50, in test_amm15_example_plot
    import amm15_example_plot
ModuleNotFoundError: No module named 'amm15_example_plot'

======================================================================
ERROR: test_anchor_plots_of_nsea_wvel (test_example_scripts.test_example_scripts)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vkamino/work/jelt/GitHub/COAsT/unit_testing/test_example_scripts.py", line 56, in test_anchor_plots_of_nsea_wvel
    import anchor_plots_of_nsea_wvel
ModuleNotFoundError: No module named 'anchor_plots_of_nsea_wvel'

======================================================================
ERROR: test_blz_example_plot (test_example_scripts.test_example_scripts)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vkamino/work/jelt/GitHub/COAsT/unit_testing/test_example_scripts.py", line 62, in test_blz_example_plot
    import blz_example_plot
ModuleNotFoundError: No module named 'blz_example_plot'

======================================================================
ERROR: test_internal_tide_pycnocline_diagnostics (test_example_scripts.test_example_scripts)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vkamino/work/jelt/GitHub/COAsT/unit_testing/test_example_scripts.py", line 80, in test_internal_tide_pycnocline_diagnostics
    import internal_tide_pycnocline_diagnostics
ModuleNotFoundError: No module named 'internal_tide_pycnocline_diagnostics'

======================================================================
ERROR: test_seasia_r12_example_plot (test_example_scripts.test_example_scripts)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vkamino/work/jelt/GitHub/COAsT/unit_testing/test_example_scripts.py", line 68, in test_seasia_r12_example_plot
    import seasia_r12_example_plot
ModuleNotFoundError: No module named 'seasia_r12_example_plot'

======================================================================
ERROR: test_wcssp_india_example_plot (test_example_scripts.test_example_scripts)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vkamino/work/jelt/GitHub/COAsT/unit_testing/test_example_scripts.py", line 74, in test_wcssp_india_example_plot
    import wcssp_india_example_plot
ModuleNotFoundError: No module named 'wcssp_india_example_plot'

======================================================================
ERROR: test_load_WOD (test_WOD_read_data.test_WOD_read_data) [Load profile data from WOD]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vkamino/work/jelt/GitHub/COAsT/unit_testing/test_WOD_read_data.py", line 21, in test_load_WOD
    WOD_profile_1D = coast.Profile(config=files.fn_WOD_config)
AttributeError: module 'unit_test_files' has no attribute 'fn_WOD_config'

======================================================================
ERROR: test_reshape_WOD (test_WOD_read_data.test_WOD_read_data)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vkamino/work/jelt/GitHub/COAsT/unit_testing/test_WOD_read_data.py", line 28, in test_reshape_WOD
    WOD_profile_1D = coast.Profile(config=files.fn_WOD_config)
AttributeError: module 'unit_test_files' has no attribute 'fn_WOD_config'

======================================================================
FAIL: test_bad_method (test_TEMPLATE.test_TEMPLATE)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vkamino/work/jelt/GitHub/COAsT/unit_testing/test_TEMPLATE.py", line 51, in test_bad_method
    self.assertTrue(1 == 2, "1 does not equal 2")
AssertionError: False is not true : 1 does not equal 2

======================================================================
FAIL: test_with_subtests (test_TEMPLATE.test_TEMPLATE) [Third subtest]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vkamino/work/jelt/GitHub/COAsT/unit_testing/test_TEMPLATE.py", line 70, in test_with_subtests
    self.assertAlmostEqual(a, c, msg="0 is not almost equal to 50000000")
AssertionError: 1 != 50000000 within 7 places (49999999 difference) : 0 is not almost equal to 50000000

----------------------------------------------------------------------
Ran 68 tests in 333.034s

FAILED (failures=2, errors=8)
<Figure size 432x288 with 0 Axes>
```

jpolton commented:
The example_script errors are fixed in a mod to `update requirements`

katavouta commented:
> 
Hi Jeff do you mean adding something like that:
   def test_example_WOD_BGC_ragged(self):
        if "livljobs" in gethostname().lower():
            import example_WOD_BGC_ragged.py

            plt.close("all")

jpolton commented:
> Hi Jeff do you mean adding something like that: def test_example_WOD_BGC_ragged(self): if "livljobs" in gethostname().lower(): import example_WOD_BGC_ragged.py
> 
> ```
>         plt.close("all")
> ```

No I think `import amm15_example_plot` needs to be `from example_scripts import amm15_example_plot` (Got to dash. Will try again later)

davbyr commented:
All of the auto checks pass now on github. There was an error in test_index where it was looking for an object called `profile` but it should have been `WOD_profile_1D` (line 83)

katavouta commented:
> All of the auto checks pass now on github. There was an error in test_index where it was looking for an object called `profile` but it should have been `WOD_profile_1D` (line 83)

Great David thank you

katavouta commented:
added example_WOD_BGC_ragged.py to unit_testing/test_example_scripts.py

davbyr commented:
@katavouta I cannot figure out why this is not passing the tests. What did you change in this most recent commit?

jpolton commented:
All good except:
```
ok
test_contour_tutorial (test_example_scripts.test_example_scripts) ... ok
test_example_WOD_BGC_ragged (test_example_scripts.test_example_scripts) ... /vkamino/work/jelt/GitHub/COAsT/coast/logging_util.py:78: UserWarning: /vkamino/work/jelt/GitHub/COAsT/coast/profile.py.__init__.41: Object created but no file or directory specified: 
<coast.profile.Profile object at 0x7f9904302430>
  return warnings.warn(add_info(msg), *args, **kwargs)
ERROR
test_export_to_netcdf_tutorial (test_example_scripts.test_example_scripts) ... ok
```

jpolton commented:
> @katavouta I cannot figure out why this is not passing the tests. What did you change in this most recent commit?

I fixed this by rerunning the action a few times

katavouta commented:
> > @katavouta I cannot figure out why this is not passing the tests. What did you change in this most recent commit?
 In the latest commits I have changed: a) the path where the data are in livjobs so path_examples = "/projectsa/COAsT/NEMO_example_data/SEAsia_R12/" inside example_WOD_BGC_ragged.py

(b) added the     def test_example_WOD_BGC_ragged(self): .... in the test_example_scripts.py

jpolton commented:
> > > @katavouta I cannot figure out why this is not passing the tests. What did you change in this most recent commit?
> > > In the latest commits I have changed: a) the path where the data are in livjobs so path_examples = "/projectsa/COAsT/NEMO_example_data/SEAsia_R12/" inside example_WOD_BGC_ragged.py
> 
> (b) added the def test_example_WOD_BGC_ragged(self): .... in the test_example_scripts.py

@katavouta can you push this back to the repo. I will test again tonight.

katavouta commented:
> > > > @katavouta I cannot figure out why this is not passing the tests. What did you change in this most recent commit?
> > > > In the latest commits I have changed: a) the path where the data are in livjobs so path_examples = "/projectsa/COAsT/NEMO_example_data/SEAsia_R12/" inside example_WOD_BGC_ragged.py
> > 
> > 
> > (b) added the def test_example_WOD_BGC_ragged(self): .... in the test_example_scripts.py
> 
> @katavouta can you push this back to the repo. I will test again tonight.

Yes I will pull and push back (thanks)

katavouta commented:
I merged 406/lower-case to it and just fixed/updated some issues with the names of files. I also added back the unit_test_BGC.py just because we have not incorporated into the new unit test (I am going to work on this now). Files changed:
#	modified:   config/example_wod_profiles.json
#	modified:   example_scripts/seasia_dic_example_plot.py
#	modified:   example_scripts/wod_bgc_ragged_example.py
#	modified:   unit_testing/test_example_scripts.py
#	modified:   unit_testing/test_index_classes_load_datasets.py
#	new file:   unit_testing/unit_test_BGC.py
#	modified:   unit_testing/unit_test_contents.txt

katavouta commented:
Maybe hold on on checking I am doing some further changes in the names to have then all lower case

katavouta commented:
Updated paths in my example files to be consistent with the other examples as having netcdf files in different directories caused problems in the unit_test. I added the unit_test_BGC in the new unit test now so unit_test_BGC can be removed after checking. 

I still however get some error in the unit_test which I am not sure where they come from. Here is the output of the unit test:

test_gridded_calculate_depth0_for_tuvwf (test_gridded_initialisation.test_gridded_initialisation) ... ok
test_gridded_compute_e3_from_ssh (test_gridded_initialisation.test_gridded_initialisation) ... ok
test_gridded_load_dimensions_correctly_renamed (test_gridded_initialisation.test_gridded_initialisation) ... ok
test_gridded_load_domain_only (test_gridded_initialisation.test_gridded_initialisation) ... ok
test_gridded_load_multiple (test_gridded_initialisation.test_gridded_initialisation) ... ok
test_gridded_load_of_data_and_domain (test_gridded_initialisation.test_gridded_initialisation) ... ok
test_gridded_load_of_data_only (test_gridded_initialisation.test_gridded_initialisation) ... /vkamino/work/annkat/COAST_DEV/COAsT/coast/logging_util.py:78: UserWarning: /vkamino/work/annkat/COAST_DEV/COAsT/coast/gridded.py._setup_grid_obj.79: No NEMO domain specified, only limited functionality will be available
  return warnings.warn(add_info(msg), *args, **kwargs)
ok
test_gridded_load_subregion_with_domain (test_gridded_initialisation.test_gridded_initialisation) ... ok
test_gridded_load_variables_correctly_renamed (test_gridded_initialisation.test_gridded_initialisation) ... ok
test_bst_to_gmt (test_general_utils.test_general_utils) ... ok
test_coast_variable_renaming (test_general_utils.test_general_utils) ... ok
test_copy_coast_object (test_general_utils.test_general_utils) ... ok
test_day_of_week (test_general_utils.test_general_utils) ... ok
test_getitem (test_general_utils.test_general_utils) ... ok
test_basic_conversion_to_xesmf (test_xesmf_convert.test_xesmf_convert) ... ok
test_combine_and_convert_harmonics (test_gridded_harmonics.test_gridded_harmonics) ... ok
test_compute_vertical_spatial_derivative (test_diagnostic_methods.test_diagnostic_methods) ... ok
test_construct_density (test_diagnostic_methods.test_diagnostic_methods) ... ok
test_construct_pycnocline_depth_and_thickness (test_diagnostic_methods.test_diagnostic_methods) ... /work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/core/computation.py:727: RuntimeWarning: invalid value encountered in sqrt
  result_data = func(*input_data)
ok
test_calculate_transport_velocity_and_depth (test_transect_methods.test_transect_methods) ... ok
test_cross_transect_geostrophic_flow (test_transect_methods.test_transect_methods) ... ok
test_determine_extract_transect_indices (test_transect_methods.test_transect_methods) ... ok
test_transect_density_and_pressure (test_transect_methods.test_transect_methods) ... ok
test_indices_by_distance (test_object_manipulation.test_object_manipulation) ... ok
test_interpolation_to_altimetry (test_object_manipulation.test_object_manipulation) ... Altimetry object at 0x56459b78bf60 initialised
Subsetting Altimetry object at 0x56459b78bf60 indices in [-10, 10], [45, 60]
ok
test_subset_single_variable (test_object_manipulation.test_object_manipulation) ... ok
test_altimetry_load_subset_and_comparison (test_altimetry_methods.test_altimetry_methods) ... Altimetry object at 0x56459b78bf60 initialised
Subsetting Altimetry object at 0x56459b78bf60 indices in [-10, 10], [45, 60]
Interpolating Gridded object at 0x56459b78bf60 "ssh" with time_interp "nearest"
Altimetry object at 0x56459b78bf60 initialised
/vkamino/work/annkat/COAST_DEV/COAsT/coast/logging_util.py:78: UserWarning: /vkamino/work/annkat/COAST_DEV/COAsT/coast/altimetry.py.__init__.67: Object created but no file or directory specified: 
<coast.altimetry.Altimetry object at 0x7f22ebff0910>
  return warnings.warn(add_info(msg), *args, **kwargs)
Altimetry object at 0x56459b78bf60 initialised
/vkamino/work/annkat/COAST_DEV/COAsT/coast/logging_util.py:78: UserWarning: /vkamino/work/annkat/COAST_DEV/COAsT/coast/altimetry.py.__init__.67: Object created but no file or directory specified: 
<coast.altimetry.Altimetry object at 0x7f21a42d7b20>
  return warnings.warn(add_info(msg), *args, **kwargs)
ok
test_load_multiple_tidegauge (test_tidegauge_methods.test_tidegauge_methods) ... Tidegauge object at 0x56459b78bf60 initialised
Tidegauge object at 0x56459b78bf60 initialised
Tidegauge object at 0x56459b78bf60 initialised
ok
test_read_gesla_and_compare_to_model (test_tidegauge_methods.test_tidegauge_methods) ... Tidegauge object at 0x56459b78bf60 initialised
Tidegauge object at 0x56459b78bf60 initialised
Tidegauge object at 0x56459b78bf60 initialised
ok
test_tidegauge_cubic_spline_extrema (test_tidegauge_methods.test_tidegauge_methods) ... Tidegauge object at 0x56459b78bf60 initialised
Tidegauge object at 0x56459b78bf60 initialised
ok
test_tidegauge_finding_extrema (test_tidegauge_methods.test_tidegauge_methods) ... ok
test_tidegauge_for_tabulated_data (test_tidegauge_methods.test_tidegauge_methods) ... Tidegauge object at 0x56459b78bf60 initialised
Tidegauge object at 0x56459b78bf60 initialised
Tidegauge object at 0x56459b78bf60 initialised
ok
test_tidegauge_resample_and_apply_doodsonx0 (test_tidegauge_methods.test_tidegauge_methods) ... Tidegauge object at 0x56459b78bf60 initialised
ok
test_compute_heofs (test_eof_methods.test_eof_methods) ... ok
test_compute_regular_eof (test_eof_methods.test_eof_methods) ... ok
test_calculate_flow_across_contour (test_isobath_contour_methods.test_contour_f_methods) ... ok
test_calculate_pressure_gradient_driven_flow (test_isobath_contour_methods.test_contour_f_methods) ... ok
test_extract_isobath_contour_between_two_points (test_isobath_contour_methods.test_contour_f_methods) ... ok
test_along_contour_2d_flow (test_isobath_contour_methods.test_contour_t_methods) ... ok
test_along_contour_flow (test_isobath_contour_methods.test_contour_t_methods) ... ok
test_calculate_pressure_along_contour (test_isobath_contour_methods.test_contour_t_methods) ... ok
test_compare_processed_profile_with_model (test_profile_methods.test_profile_methods) ... ERROR
test_load_process_and_compare_profile_data (test_profile_methods.test_profile_methods) ... test_determine_clim_by_stdev (test_plot_utilities.test_plot_utilities) ... ok
test_determine_colorbar_extension (test_plot_utilities.test_plot_utilities) ... ok
test_geo_axes (test_plot_utilities.test_plot_utilities) ... ok
test_scatter_with_fit (test_plot_utilities.test_plot_utilities) ... ok
test_find_maxima (test_stats_utilities.test_stats_utilities) ... ok
test_fill_polygon_by_index (test_maskmaker_methods.test_maskmaker_methods) ... Tidegauge object at 0x56459b78bf60 initialised
ok
test_fill_polygon_by_lonlat (test_maskmaker_methods.test_maskmaker_methods) ... ok
[########################################] | 100% Completed |  0.6sology) ... 
/vkamino/work/annkat/COAST_DEV/COAsT/./unit_testing/test_climatology.py:32: RuntimeWarning: Mean of empty slice
  mn = mn = np.nanmean(ds.temperature, axis=0)
ok
test_load_wod (test_wod_read_data.test_wod_read_data) ... /vkamino/work/annkat/COAST_DEV/COAsT/coast/logging_util.py:78: UserWarning: /vkamino/work/annkat/COAST_DEV/COAsT/coast/profile.py.__init__.41: Object created but no file or directory specified: 
<coast.profile.Profile object at 0x7f208303c4f0>
  return warnings.warn(add_info(msg), *args, **kwargs)
ok
test_reshape_wod (test_wod_read_data.test_wod_read_data) ... /vkamino/work/annkat/COAST_DEV/COAsT/coast/logging_util.py:78: UserWarning: /vkamino/work/annkat/COAST_DEV/COAsT/coast/profile.py.__init__.41: Object created but no file or directory specified: 
<coast.profile.Profile object at 0x7f218433e460>
  return warnings.warn(add_info(msg), *args, **kwargs)
Depth OK reshape successful
dic
observed variable exist
OK reshape successful
temperature
observed variable exist
OK reshape successful
alkalinity
variable not in observations
/vkamino/work/annkat/COAST_DEV/COAsT/coast/logging_util.py:78: UserWarning: /vkamino/work/annkat/COAST_DEV/COAsT/coast/profile.py.__init__.41: Object created but no file or directory specified: 
<coast.profile.Profile object at 0x7f218430b910>
  return warnings.warn(add_info(msg), *args, **kwargs)
ok
test_gridded_load_bgc_data (test_bgc_gridded_initialisation.test_bgc_gridded_initialisation) ... /vkamino/work/annkat/COAST_DEV/COAsT/coast/logging_util.py:78: UserWarning: /vkamino/work/annkat/COAST_DEV/COAsT/coast/gridded.py._setup_grid_obj.79: No NEMO domain specified, only limited functionality will be available
  return warnings.warn(add_info(msg), *args, **kwargs)
ok
test_gridded_load_bgc_data_and_domain (test_bgc_gridded_initialisation.test_bgc_gridded_initialisation) ... ok
test_gridded_load_bgc_dimensions_correctly_renamed (test_bgc_gridded_initialisation.test_bgc_gridded_initialisation) ... ok
test_gridded_load_bgc_domain_only (test_bgc_gridded_initialisation.test_bgc_gridded_initialisation) ... ok
test_gridded_load_bgc_variables_correctly_renamed (test_bgc_gridded_initialisation.test_bgc_gridded_initialisation) ... ok
test_altimetry_tutorial (test_example_scripts.test_example_scripts) ... Altimetry object at 0x56459b78bf60 initialised
Subsetting Altimetry object at 0x56459b78bf60 indices in [-10, 10], [45, 60]
Interpolating Gridded object at 0x56459b78bf60 "ssh" with time_interp "nearest"
Altimetry object at 0x56459b78bf60 initialised
/vkamino/work/annkat/COAST_DEV/COAsT/coast/logging_util.py:78: UserWarning: /vkamino/work/annkat/COAST_DEV/COAsT/coast/altimetry.py.__init__.67: Object created but no file or directory specified: 
<coast.altimetry.Altimetry object at 0x7f2083194670>
  return warnings.warn(add_info(msg), *args, **kwargs)
Altimetry object at 0x56459b78bf60 initialised
/vkamino/work/annkat/COAST_DEV/COAsT/coast/logging_util.py:78: UserWarning: /vkamino/work/annkat/COAST_DEV/COAsT/coast/altimetry.py.__init__.67: Object created but no file or directory specified: 
<coast.altimetry.Altimetry object at 0x7f2082dc1730>
  return warnings.warn(add_info(msg), *args, **kwargs)
ok
test_amm15_example_plot (test_example_scripts.test_example_scripts) ... /vkamino/work/annkat/COAST_DEV/COAsT/coast/gridded.py:181: UserWarning: The model domain loaded, '/projectsa/NEMO/gmaya/AMM15_GRID/amm15.mesh_mask.cs3x.nc', does not contain the bathy_metry' variable. This will result in the NEMO.dataset.bathymetry variable being set to zero, which may result in unexpected behaviour from routines that require this variable.
  warnings.warn(
* Loaded  /work/jelt/GitHub/COAsT/config/example_nemo_grid_t.json  data
* Extract North Sea subdomain
/vkamino/work/annkat/COAST_DEV/COAsT/example_scripts/amm15_example_plot.py:66: UserWarning: The input coordinates to pcolormesh are interpreted as cell centers, but are not monotonically increasing or decreasing. This may lead to incorrectly calculated cell edges, in which case, please supply explicit cell edges to pcolormesh.
  plt.pcolormesh(sci_t.dataset.longitude, sci_t.dataset.latitude, sci_t.dataset.temperature.isel(z_dim=0).squeeze())
ok
test_anchor_plots_of_nsea_wvel (test_example_scripts.test_example_scripts) ... /work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/plugins.py:110: RuntimeWarning: 'netcdf4' fails while guessing
  warnings.warn(f"{engine!r} fails while guessing", RuntimeWarning)
/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/plugins.py:110: RuntimeWarning: 'scipy' fails while guessing
  warnings.warn(f"{engine!r} fails while guessing", RuntimeWarning)
/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/plugins.py:119: RuntimeWarning: 'netcdf4' fails while guessing
  warnings.warn(f"{engine!r} fails while guessing", RuntimeWarning)
/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/plugins.py:119: RuntimeWarning: 'h5netcdf' fails while guessing
  warnings.warn(f"{engine!r} fails while guessing", RuntimeWarning)
/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/plugins.py:119: RuntimeWarning: 'scipy' fails while guessing
  warnings.warn(f"{engine!r} fails while guessing", RuntimeWarning)
ERROR
test_blz_example_plot (test_example_scripts.test_example_scripts) ... /vkamino/work/annkat/COAST_DEV/COAsT/coast/gridded.py:181: UserWarning: The model domain loaded, '/projectsa/accord/GCOMS1k/INPUTS/BLZE12_C1/domain_cfg.nc', does not contain the bathy_metry' variable. This will result in the NEMO.dataset.bathymetry variable being set to zero, which may result in unexpected behaviour from routines that require this variable.
  warnings.warn(
/vkamino/work/annkat/COAST_DEV/COAsT/coast/gridded.py:181: UserWarning: The model domain loaded, '/projectsa/accord/GCOMS1k/INPUTS/BLZE12_C1/domain_cfg.nc', does not contain the bathy_metry' variable. This will result in the NEMO.dataset.bathymetry variable being set to zero, which may result in unexpected behaviour from routines that require this variable.
  warnings.warn(
/vkamino/work/annkat/COAST_DEV/COAsT/coast/gridded.py:181: UserWarning: The model domain loaded, '/projectsa/accord/GCOMS1k/INPUTS/BLZE12_C1/domain_cfg.nc', does not contain the bathy_metry' variable. This will result in the NEMO.dataset.bathymetry variable being set to zero, which may result in unexpected behaviour from routines that require this variable.
  warnings.warn(
ok
test_contour_tutorial (test_example_scripts.test_example_scripts) ... <Figure size 432x288 with 0 Axes>
ok
test_export_to_netcdf_tutorial (test_example_scripts.test_example_scripts) ... ok
test_internal_tide_pycnocline_diagnostics (test_example_scripts.test_example_scripts) ... /work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/plugins.py:110: RuntimeWarning: 'netcdf4' fails while guessing
  warnings.warn(f"{engine!r} fails while guessing", RuntimeWarning)
/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/plugins.py:110: RuntimeWarning: 'scipy' fails while guessing
  warnings.warn(f"{engine!r} fails while guessing", RuntimeWarning)
/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/plugins.py:119: RuntimeWarning: 'netcdf4' fails while guessing
  warnings.warn(f"{engine!r} fails while guessing", RuntimeWarning)
/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/plugins.py:119: RuntimeWarning: 'h5netcdf' fails while guessing
  warnings.warn(f"{engine!r} fails while guessing", RuntimeWarning)
/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/plugins.py:119: RuntimeWarning: 'scipy' fails while guessing
  warnings.warn(f"{engine!r} fails while guessing", RuntimeWarning)
* Loaded  AMM7  data
* Extract North Sea subdomain
* Construct in-situ density and stratification
* Construct stratification. t-pts --> w-pts
* Create internal tide diagnostics object
* Compute density and rho_dz if they didnt exist
* Compute 1st and 2nd moments of stratification as pycnocline vars
* Sample quick plot
/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/core/computation.py:727: RuntimeWarning: invalid value encountered in sqrt
  result_data = func(*input_data)
* Construct transects to inspect stratification. This is an abuse of the transect code...
 - I have forced the w-pt nemo data and w-pt IT data into the t-point Transect objects

* Plot sections with pycnocline depth and thickness overlayed
/vkamino/work/annkat/COAST_DEV/COAsT/example_scripts/internal_tide_pycnocline_diagnostics.py:149: UserWarning: The input coordinates to pcolormesh are interpreted as cell centers, but are not monotonically increasing or decreasing. This may lead to incorrectly calculated cell edges, in which case, please supply explicit cell edges to pcolormesh.
  plt.pcolormesh(lat_sec, dep_sec, rho_sec)
/vkamino/work/annkat/COAST_DEV/COAsT/example_scripts/internal_tide_pycnocline_diagnostics.py:158: UserWarning: The input coordinates to pcolormesh are interpreted as cell centers, but are not monotonically increasing or decreasing. This may lead to incorrectly calculated cell edges, in which case, please supply explicit cell edges to pcolormesh.
  plt.pcolormesh(lat_sec, dep_sec, strat_sec)
* Plot profile of density and stratification with strat_1st_mom in deep water
 - When the stratification is not nearly two-layer, then then there is no sharp pycnocline for the  1st and 2nd moments to pick out. You end up with a thick pycnocline and reduced precision on the depth

* Map pretty plots of North Sea pycnocline depth
 - we expect a RunTimeError here
ok
test_seasia_dic_example_plot (test_example_scripts.test_example_scripts) ... /vkamino/work/annkat/COAST_DEV/COAsT/example_scripts/seasia_dic_example_plot.py:27: UserWarning: The input coordinates to pcolormesh are interpreted as cell centers, but are not monotonically increasing or decreasing. This may lead to incorrectly calculated cell edges, in which case, please supply explicit cell edges to pcolormesh.
  plt.pcolormesh(
ok
test_seasia_r12_example_plot (test_example_scripts.test_example_scripts) ... /vkamino/work/annkat/COAST_DEV/COAsT/coast/gridded.py:181: UserWarning: The model domain loaded, '/projectsa/COAsT/NEMO_example_data/SEAsia_R12/domain_cfg_ORCA12_adj.nc', does not contain the bathy_metry' variable. This will result in the NEMO.dataset.bathymetry variable being set to zero, which may result in unexpected behaviour from routines that require this variable.
  warnings.warn(
/vkamino/work/annkat/COAST_DEV/COAsT/example_scripts/seasia_r12_example_plot.py:27: UserWarning: The input coordinates to pcolormesh are interpreted as cell centers, but are not monotonically increasing or decreasing. This may lead to incorrectly calculated cell edges, in which case, please supply explicit cell edges to pcolormesh.
  plt.pcolormesh(sci_t.dataset.longitude, sci_t.dataset.latitude, sci_t.dataset.salinity.isel(t_dim=0).isel(z_dim=0))
ok
test_tidegauge_tutorial (test_example_scripts.test_example_scripts) ... Tidegauge object at 0x56459b78bf60 initialised
Tidegauge object at 0x56459b78bf60 initialised
Tidegauge object at 0x56459b78bf60 initialised
Tidegauge object at 0x56459b78bf60 initialised
Tidegauge object at 0x56459b78bf60 initialised
Tidegauge object at 0x56459b78bf60 initialised
Tidegauge object at 0x56459b78bf60 initialised
Tidegauge object at 0x56459b78bf60 initialised
Tidegauge object at 0x56459b78bf60 initialised
ok
test_tidetable_tutorial (test_example_scripts.test_example_scripts) ... load and plot HLW data
Tidegauge object at 0x56459b78bf60 initialised
Try the TideGauge.show() method:
Try the TideGauge.get_tidetabletimes() methods:
LT: 2.83 m at 2020-10-13T14:36:00.000000000
HT: 8.01 m at 2020-10-13T07:59:00.000000000
HT: 8.01 m at 2020-10-13T07:59:00.000000000
ok
test_transect_tutorial (test_example_scripts.test_example_scripts) ... ok
test_wcssp_india_example_plot (test_example_scripts.test_example_scripts) ... /vkamino/work/annkat/COAST_DEV/COAsT/coast/gridded.py:181: UserWarning: The model domain loaded, '/projectsa/COAsT/NEMO_example_data/MO_INDIA/domain_cfg_wcssp.nc', does not contain the bathy_metry' variable. This will result in the NEMO.dataset.bathymetry variable being set to zero, which may result in unexpected behaviour from routines that require this variable.
  warnings.warn(
ok
test_wod_bgc_ragged_example (test_example_scripts.test_example_scripts) ... /vkamino/work/annkat/COAST_DEV/COAsT/coast/logging_util.py:78: UserWarning: /vkamino/work/annkat/COAST_DEV/COAsT/coast/profile.py.__init__.41: Object created but no file or directory specified: 
<coast.profile.Profile object at 0x7f21486fb820>
  return warnings.warn(add_info(msg), *args, **kwargs)
Depth OK reshape successful
salinity
observed variable exist
OK reshape successful
temperature
observed variable exist
OK reshape successful
nitrate
variable not in observations
oxygen
observed variable exist
OK reshape successful
dic
observed variable exist
OK reshape successful
phosphate
observed variable exist
OK reshape successful
alkalinity
variable not in observations
/vkamino/work/annkat/COAST_DEV/COAsT/coast/logging_util.py:78: UserWarning: /vkamino/work/annkat/COAST_DEV/COAsT/coast/profile.py.__init__.41: Object created but no file or directory specified: 
<coast.profile.Profile object at 0x7f23c0390490>
  return warnings.warn(add_info(msg), *args, **kwargs)
/vkamino/work/annkat/COAST_DEV/COAsT/coast/logging_util.py:78: UserWarning: /vkamino/work/annkat/COAST_DEV/COAsT/coast/profile.py.__init__.41: Object created but no file or directory specified: 
<coast.profile.Profile object at 0x7f20831829d0>
  return warnings.warn(add_info(msg), *args, **kwargs)
/vkamino/work/annkat/COAST_DEV/COAsT/coast/logging_util.py:78: UserWarning: /vkamino/work/annkat/COAST_DEV/COAsT/coast/profile.py.__init__.41: Object created but no file or directory specified: 
<coast.profile.Profile object at 0x7f2083194760>
  return warnings.warn(add_info(msg), *args, **kwargs)
/vkamino/work/annkat/COAST_DEV/COAsT/coast/logging_util.py:78: UserWarning: /vkamino/work/annkat/COAST_DEV/COAsT/coast/profile.py.__init__.41: Object created but no file or directory specified: 
<coast.profile.Profile object at 0x7f22ebbefbb0>
  return warnings.warn(add_info(msg), *args, **kwargs)
ok

======================================================================
ERROR: test_compare_processed_profile_with_model (test_profile_methods.test_profile_methods)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/file_manager.py", line 199, in _acquire_with_cache_info
    file = self._cache[self._key]
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/lru_cache.py", line 53, in __getitem__
    value = self._cache[key]
KeyError: [<class 'netCDF4._netCDF4.Dataset'>, ('/vkamino/work/annkat/COAST_DEV/COAsT/example_files/coast_example_EN4_201008.nc',), 'r', (('clobber', True), ('diskless', False), ('format', 'NETCDF4'), ('persist', False))]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/vkamino/work/annkat/COAST_DEV/COAsT/./unit_testing/test_profile_methods.py", line 41, in test_compare_processed_profile_with_model
    profile = coast.Profile(files.fn_profile, config=files.fn_profile_config)
  File "/vkamino/work/annkat/COAST_DEV/COAsT/coast/profile.py", line 43, in __init__
    self.read_en4(file_path, self.chunks, multiple)
  File "/vkamino/work/annkat/COAST_DEV/COAsT/coast/profile.py", line 57, in read_en4
    self.dataset = xr.open_dataset(fn_en4, chunks=chunks)
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/api.py", line 495, in open_dataset
    backend_ds = backend.open_dataset(
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/netCDF4_.py", line 553, in open_dataset
    store = NetCDF4DataStore.open(
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/netCDF4_.py", line 382, in open
    return cls(manager, group=group, mode=mode, lock=lock, autoclose=autoclose)
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/netCDF4_.py", line 330, in __init__
    self.format = self.ds.data_model
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/netCDF4_.py", line 391, in ds
    return self._acquire()
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/netCDF4_.py", line 385, in _acquire
    with self._manager.acquire_context(needs_lock) as root:
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/contextlib.py", line 113, in __enter__
    return next(self.gen)
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/file_manager.py", line 187, in acquire_context
    file, cached = self._acquire_with_cache_info(needs_lock)
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/file_manager.py", line 205, in _acquire_with_cache_info
    file = self._opener(*self._args, **kwargs)
  File "src/netCDF4/_netCDF4.pyx", line 2307, in netCDF4._netCDF4.Dataset.__init__
  File "src/netCDF4/_netCDF4.pyx", line 1925, in netCDF4._netCDF4._ensure_nc_success
FileNotFoundError: [Errno 2] No such file or directory: b'/vkamino/work/annkat/COAST_DEV/COAsT/example_files/coast_example_EN4_201008.nc'

======================================================================
ERROR: test_load_process_and_compare_profile_data (test_profile_methods.test_profile_methods) [Load profile data from EN4]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/file_manager.py", line 199, in _acquire_with_cache_info
    file = self._cache[self._key]
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/lru_cache.py", line 53, in __getitem__
    value = self._cache[key]
KeyError: [<class 'netCDF4._netCDF4.Dataset'>, ('/vkamino/work/annkat/COAST_DEV/COAsT/example_files/coast_example_EN4_201008.nc',), 'r', (('clobber', True), ('diskless', False), ('format', 'NETCDF4'), ('persist', False))]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/vkamino/work/annkat/COAST_DEV/COAsT/./unit_testing/test_profile_methods.py", line 18, in test_load_process_and_compare_profile_data
    profile = coast.Profile(files.fn_profile, config=files.fn_profile_config)
  File "/vkamino/work/annkat/COAST_DEV/COAsT/coast/profile.py", line 43, in __init__
    self.read_en4(file_path, self.chunks, multiple)
  File "/vkamino/work/annkat/COAST_DEV/COAsT/coast/profile.py", line 57, in read_en4
    self.dataset = xr.open_dataset(fn_en4, chunks=chunks)
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/api.py", line 495, in open_dataset
    backend_ds = backend.open_dataset(
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/netCDF4_.py", line 553, in open_dataset
    store = NetCDF4DataStore.open(
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/netCDF4_.py", line 382, in open
    return cls(manager, group=group, mode=mode, lock=lock, autoclose=autoclose)
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/netCDF4_.py", line 330, in __init__
    self.format = self.ds.data_model
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/netCDF4_.py", line 391, in ds
    return self._acquire()
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/netCDF4_.py", line 385, in _acquire
    with self._manager.acquire_context(needs_lock) as root:
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/contextlib.py", line 113, in __enter__
    return next(self.gen)
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/file_manager.py", line 187, in acquire_context
    file, cached = self._acquire_with_cache_info(needs_lock)
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/file_manager.py", line 205, in _acquire_with_cache_info
    file = self._opener(*self._args, **kwargs)
  File "src/netCDF4/_netCDF4.pyx", line 2307, in netCDF4._netCDF4.Dataset.__init__
  File "src/netCDF4/_netCDF4.pyx", line 1925, in netCDF4._netCDF4._ensure_nc_success
FileNotFoundError: [Errno 2] No such file or directory: b'/vkamino/work/annkat/COAST_DEV/COAsT/example_files/coast_example_EN4_201008.nc'

======================================================================
ERROR: test_load_process_and_compare_profile_data (test_profile_methods.test_profile_methods) [Process EN4 profile data]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vkamino/work/annkat/COAST_DEV/COAsT/./unit_testing/test_profile_methods.py", line 28, in test_load_process_and_compare_profile_data
    processed = profile.process_en4()
UnboundLocalError: local variable 'profile' referenced before assignment

======================================================================
ERROR: test_anchor_plots_of_nsea_wvel (test_example_scripts.test_example_scripts)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vkamino/work/annkat/COAST_DEV/COAsT/./unit_testing/test_example_scripts.py", line 56, in test_anchor_plots_of_nsea_wvel
    from example_scripts import anchor_plots_of_nsea_wvel
  File "/vkamino/work/annkat/COAST_DEV/COAsT/example_scripts/anchor_plots_of_nsea_wvel.py", line 32, in <module>
    sci_w = coast.Gridded(dir_nam + fil_nam, dom_nam, config=config)
  File "/vkamino/work/annkat/COAST_DEV/COAsT/coast/gridded.py", line 49, in __init__
    self._setup_grid_obj(None, multiple, **kwargs)
  File "/vkamino/work/annkat/COAST_DEV/COAsT/coast/gridded.py", line 82, in _setup_grid_obj
    dataset_domain = self.load_domain(self.fn_domain, chunks)
  File "/vkamino/work/annkat/COAST_DEV/COAsT/coast/gridded.py", line 110, in load_domain
    dataset_domain = xr.open_dataset(fn_domain)
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/api.py", line 479, in open_dataset
    engine = plugins.guess_engine(filename_or_obj)
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/plugins.py", line 148, in guess_engine
    raise ValueError(error_msg)
ValueError: did not find a match in any of xarray's currently installed IO backends ['netcdf4', 'scipy']. Consider explicitly selecting one of the installed engines via the ``engine`` parameter, or installing additional IO dependencies, see:
https://docs.xarray.dev/en/stable/getting-started-guide/installing.html
https://docs.xarray.dev/en/stable/user-guide/io.html

----------------------------------------------------------------------
Ran 72 tests in 628.615s

FAILED (errors=4)

katavouta commented:
fix most problems: the problem was that in the example files it searches for coast_example_EN4_201008.nc while in the download examples the name of the file is coast_example_en4_201008.nc. I did not change the code but renaming the file fixes the problems with the profile unit test

Only one error remains in the unit test:
======================================================================
ERROR: test_anchor_plots_of_nsea_wvel (test_example_scripts.test_example_scripts)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vkamino/work/annkat/COAST_DEV/COAsT/./unit_testing/test_example_scripts.py", line 56, in test_anchor_plots_of_nsea_wvel
    from example_scripts import anchor_plots_of_nsea_wvel
  File "/vkamino/work/annkat/COAST_DEV/COAsT/example_scripts/anchor_plots_of_nsea_wvel.py", line 32, in <module>
    sci_w = coast.Gridded(dir_nam + fil_nam, dom_nam, config=config)
  File "/vkamino/work/annkat/COAST_DEV/COAsT/coast/gridded.py", line 49, in __init__
    self._setup_grid_obj(None, multiple, **kwargs)
  File "/vkamino/work/annkat/COAST_DEV/COAsT/coast/gridded.py", line 82, in _setup_grid_obj
    dataset_domain = self.load_domain(self.fn_domain, chunks)
  File "/vkamino/work/annkat/COAST_DEV/COAsT/coast/gridded.py", line 110, in load_domain
    dataset_domain = xr.open_dataset(fn_domain)
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/api.py", line 479, in open_dataset
    engine = plugins.guess_engine(filename_or_obj)
  File "/work/annkat/annkat-conda/coast_dev/lib/python3.8/site-packages/xarray/backends/plugins.py", line 148, in guess_engine
    raise ValueError(error_msg)
ValueError: did not find a match in any of xarray's currently installed IO backends ['netcdf4', 'scipy']. Consider explicitly selecting one of the installed engines via the ``engine`` parameter, or installing additional IO dependencies, see:
https://docs.xarray.dev/en/stable/getting-started-guide/installing.html
https://docs.xarray.dev/en/stable/user-guide/io.html

----------------------------------------------------------------------
Ran 72 tests in 621.040s

FAILED (errors=1)

davbyr commented:
@katavouta Glad you fixed the problem with the EN4 file. When did the name of the en4 file get capitalized? Was that from the merge with Jeff's branch?

Another example script error.... hm. We need to be careful with including these in the unit tests with their hard coded paths etc.

katavouta commented:
> @katavouta Glad you fixed the problem with the EN4 file. When did the name of the en4 file get capitalized? Was that from the merge with Jeff's branch?
> 
 I am not sure what happens when I download the examples is that the actual file is called coast_example_en4_201008.nc, but then I think there is automatically a virtual link when you unzip the files as: coast_example_EN4_201008.nc -> coast_example_en4_201008.nc

However when I then create my virtual links like for them to be inside the ./example_files in my COAST directory it does not pick up this virtual link directly, so it did not work. I had to create a link and then it worked fine. just something to be aware of

katavouta commented:
> @katavouta Glad you fixed the problem with the EN4 file. When did the name of the en4 file get capitalized? Was that from the merge with Jeff's branch?
> 
Also if you look in the develop branch you will see that in the unit_test_files.py:
fn_profile = path.join(dn_files, "coast_example_EN4_201008.nc")

so I think it may originally have being like that and that is what caused the problem

davbyr reviewed (approved):
Unit test works for me locally

```
----------------------------------------------------------------------
Ran 72 tests in 278.179s

OK
```
